src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -113,22 +111,18 @@ object PeerDiscoveryManager {
             nodeStatusHolder: Agent[NodeStatus]): Props =
     Props(new PeerDiscoveryManager(discoveryListener, discoveryConfig, knownNodesStorage, nodeStatusHolder))
 
-  object Node {
+  object NodeWithTimestamp {
 
-    def fromUri(uri: URI): Node = {
-      val nodeId = ByteString(Hex.decode(uri.getUserInfo))
-      Node(nodeId, new InetSocketAddress(uri.getHost, uri.getPort), System.currentTimeMillis())
-    }
-  }
+    def fromUri(uri: URI): NodeWithTimestamp = fromNode(NodeImpl.fromUri(uri))
+
+    def fromNode(node: Node): NodeWithTimestamp = NodeWithTimestamp(node.id, node.addr, System.currentTimeMillis())
```

Comment:
to be honest, I'd really prefer to use just a single `case class Node()` (without splitting it into trait and impl). And use that class here, like: `DiscoveryNodeInfo(node: Node, timestamp)`, alternatively we could add another map: `addTimestamps: Map[NodeId, timestamp]` and just use the `Node` in original map.